azure/scope/machine_test.go

```diff
@@ -223,3 +229,829 @@ func TestMachineScope_ProviderID(t *testing.T) {
+			machineScope: MachineScope{
+				AzureMachine: &infrav1.AzureMachine{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "machine-name",
+					},
+					Spec: infrav1.AzureMachineSpec{
+						SubnetName: "machine-name-subnet",
+					},
+				},
+				ClusterScoper: &ClusterScope{
+					AzureCluster: &infrav1.AzureCluster{
+						Spec: infrav1.AzureClusterSpec{
+							NetworkSpec: infrav1.NetworkSpec{
+								Subnets: []infrav1.SubnetSpec{
+									{
+										Name: "machine-name-subnet",
+									},
+									{
+										Name: "another-machine-name-subnet",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			want: infrav1.SubnetSpec{
+				Name: "machine-name-subnet",
+			},
+		},
+		{
+			name: "returns empty if machine subnet name is not present in the cluster scope",
+			machineScope: MachineScope{
+				AzureMachine: &infrav1.AzureMachine{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "machine-name",
+					},
+					Spec: infrav1.AzureMachineSpec{
+						SubnetName: "machine-name-subnet",
+					},
+				},
+				ClusterScoper: &ClusterScope{
+					AzureCluster: &infrav1.AzureCluster{
+						Spec: infrav1.AzureClusterSpec{
+							NetworkSpec: infrav1.NetworkSpec{
+								Subnets: []infrav1.SubnetSpec{
+									{
+										Name: "machine-name-subnet",
+									},
+									{
+										Name: "another-machine-name-subnet",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			want: infrav1.SubnetSpec{
+				Name: "machine-name-subnet",
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := tt.machineScope.Subnet(); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("Subnet() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestMachineScope_AvailabilityZone(t *testing.T) {
+	tests := []struct {
+		name         string
+		machineScope MachineScope
+		want         string
+	}{
+		{
+			name: "returns empty if no failure domain is present",
+			machineScope: MachineScope{
+				Machine: &clusterv1.Machine{
+					Spec: clusterv1.MachineSpec{},
+				},
+				AzureMachine: &infrav1.AzureMachine{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "machine-name",
+					},
+					Spec: infrav1.AzureMachineSpec{},
+				},
+			},
+			want: "",
+		},
+		{
+			name: "returns failure domain from the machine spec",
+			machineScope: MachineScope{
+				Machine: &clusterv1.Machine{
+					Spec: clusterv1.MachineSpec{
+						FailureDomain: pointer.String("dummy-failure-domain-from-machine-spec"),
+					},
+				},
+				AzureMachine: &infrav1.AzureMachine{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "machine-name",
+					},
+					Spec: infrav1.AzureMachineSpec{
+						FailureDomain: pointer.String("dummy-failure-domain-from-azuremachine-spec"),
+					},
+				},
+			},
+			want: "dummy-failure-domain-from-machine-spec",
+		},
+		{
+			name: "returns failure domain from the azuremachine spec",
```

Comment:
is this test case a duplicate from the one above?

Reply:
No. If failure domain is specified in both `MachineSpec` as well as `AzureMachineSpec` than the function returns the failure domain from `MachineSpec`. 
In this test case failure domain is not specified on `MachineSpec` and in above test case failure domain is specified on both `MachineSpec` as well as `AzureMachineSpec`